### DISCUSSION TITLE
关于gs使用疑问

### DISCUSSION THREAD

**Discussion by wzg-zhuo** [alibaba/GraphScope#2396] [Q&A] [answered]
使用gs最近有以下问题，还请帮忙解答：
1请问gs单机模式下，启动的进程数量默认是多少，如何实现单机多workers操作？
2单机是否创建了默认的session，创建的默认session参数在哪里可以看到？
3集群模式下是否强依赖k8s，每个session是否可以载入多个图，并进行图融合、差异对比等操作
4gs和flink-gelly是否有性能上的测试比较可以参考？

**Comment by sighingnow** [comment, marked as answer]
Hi @wzg-zhuo,

对于您的问题，

1. default session是1，`graphscope.session()` 是 2，可以 `graphscope.session(num_workers=4)` 来实现更多worker
2. default session需要用到的才会创建，例如 `graphscope.g()` 时
   - default session有接口可以查询和获取：https://github.com/alibaba/GraphScope/blob/main/python/graphscope/client/session.py#L1435-L1449
   - 用户的session也可以被设置为default session：https://github.com/alibaba/GraphScope/blob/main/python/graphscope/client/session.py#L913
3. 目前是的，依赖kubernetes。每个session可以载入多个图的，例如 `g1 = sess.g(); g2 = sess.g()`
4. 和flink-gelly没有，但是有和别的更快的图分析系统的对比：https://github.com/alibaba/GraphScope/blob/main/docs/performance_and_benchmark.md
